i2c/bmp280_i2c/bmp280_i2c.c

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright (c) 2021 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ **/
+
+#include <stdio.h>
+
+#include "hardware/i2c.h"
+#include "pico/binary_info.h"
+#include "pico/stdlib.h"
+
+/* Example code to talk to a BMP280 temperature and pressure sensor
+
+   NOTE: Ensure the device is capable of being driven at 3.3v NOT 5v. The Pico
+   GPIO (and therefore I2C) cannot be used at 5v.
+
+   You will need to use a level shifter on the I2C lines if you want to run the
+   board at 5v.
+
+   Connections on Raspberry Pi Pico board, other boards may vary.
+
+   GPIO PICO_DEFAULT_I2C_SDA_PIN (on Pico this is 4 (pin 6)) -> SDA on BMP280
+   board GPIO PICO_DEFAULT_I2C_SCK_PIN (on Pico this is 5 (pin 7)) -> SCL on
+   BMP280 board 3.3v (pin 36) -> VCC on BMP280 board GND (pin 38)  -> GND on
+   BMP280 board
+*/
+
+// number of calibration registers to be read
+#define NUM_CALIB_PARAMS 24
+
+// device has default bus address of 0x76
+const uint8_t addr = 0x76;
```

Comment:
I don't feel strongly, but curious why you chose `const uin8_t` over `#defines` for these

Reply:
I wasn't sure how to use `#define`s and hex numbers. I think there's a macro but I wasn't sure..

Reply:
Yeah, we use `_u(3)` or whatever in the SDK in case the constant is included by assembly, which doesn't understand `3u` - i.e. we define `_u` to either suffix `u` or not based on where we are.

Note also, we tend to use `u` suffixes directly for shifts etc because higher warning levels complain otherwise, but i shut up about that because again `3u` may confuse some examplees, and so we don't require quietness on such high warning levels for the examples

Reply:
I've changed to #defines so that it's consistent with other examples I'll be uploading from now.